docs/builtin-objects.md

```diff
@@ -0,0 +1,50 @@
+# helmfile template built-in objects
+
+- `Environment`: The information about the environment. This is set by the
+  `--environment` flag. It has several objects inside of it:
+  - `Environment.Name`: The name of the environment
+- `StateFile`: The information about the state file. It has several objects
+  inside of it:
+  - `StateFile.Name`: The name of the current state file
+  - `StateFile.BasePath`: The base path of the current state file
```

Comment:
Sorry for the back-and-forth but I come up with a potentially better API:

```suggestion
- `Path`: The file path to the current helmfile. 
  - `Path.Dir`: The name of the directory in which the current helmfile resides
  - `Path.Base`: The file name, or the last element of the path of the current helmfile file
```

Also, I think "StateFile.BasePath" a.k.a "Path.Dir" can be a security hole if the attacker uses it to grab information about the directory structure of the helmfile's execution environment.

How would you use `StateFile.BasePath`? An example helmfile.yaml would help!

Reply:
@mumoshu  No worries. the new name is better. it's easy to update the code. I will do that today.